### PR TITLE
Add zeroRe()

### LINKED ISSUE #456: Feature Request: Zero out residuals
If we wish to align with `mrgsolve`, probably called `zeroRe()`, though it can be something else.

Feature request:
- Ability to zero out between subject variability overall (can select, `omega` or `sigma`-like)
- Possibly ability to provide arbitrary parameters and zero/fix them, which means it could be for `theta`, `omega` or `sigma` like parameters

@billdenney  had some code for this

### PR DESCRIPTION
**Pull request by billdenney**
Fix #456

### CODE REVIEW

**Comment by billdenney** [comment]
@mattfidler, The only other two parts that I thought of while writing this are:

* Should there also be an option to fix the parameters?  I think that the way that zeros are handled for these is that they are implicitly fixed or dropped, but maybe that needs to be made explicit.
* Is there any need for boundary value checking?

File: R/piping-ini.R [hunk 1]
```diff
@@ -549,3 +549,65 @@ ini.default <- function(x, ..., envir=parent.frame(), append = NULL) {
...
+  # In the code below there is no test for bounds since the bounds are typically (0, Inf).
+  if ("omega" %in% which) {
+    omegaNames <- .ret$iniDf$name[!is.na(.ret$iniDf$neta1)]
+    if (length(omegaNames) == 0) {
+      cli::cli_warn("No omega parameters in the model")
+    } else {
+      iniCommand <- c(iniCommand, paste0(omegaNames, "~0"))
+    }
+  }
+  if ("sigma" %in% which) {
```
[the remark this answers is not in the document]
Comment (mattfidler) [reply R/piping-ini.R:596]: I'm unsure if it simulates at the boundary, but it definately will have a hard time estimating.  Maybe this is ok...

File: tests/testthat/test-piping-ini.R [hunk 2]
```diff
@@ -316,3 +316,120 @@ test_that("ini tests for different types of expressions", {
...
+  )
+  expect_equal(newMod$iniDf, newUi$iniDf)
+  # detect no change
+  expect_equal(uiNone$iniDf$est, 1)
+  expect_equal(newMod$iniDf$est, 1)
+
+  # expected errors
+  expect_error(zeroRe("A"), regexp = "'object' needs to be a rxUi model")
+  expect_error(zeroRe(modOmegaSigma, which = "foo"), regexp = "should be one of")
+})
```
Comment (mattfidler) [inline tests/testthat/test-piping-ini.R:435]: Probably should add at least one test of solving the system.  Probably one-cmt simulate.

**Comment by mattfidler** [comment]
Thanks @billdenney 

As for your questison:

- Maybe the should be fixed by default in the inital estimates?  with a flag?
- Boundaries should be checked and what is done should be complained about 
- At least one simulation run in check (ideally, I'm not sure it runs with simply reading this commit).

File: R/piping-ini.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (billdenney) [reply R/piping-ini.R:596]: I confirmed that the lower bound for `add.sd` is set to 0 when it is set to zero in the initial model.  So, this is consistent.  I have added bound checking to ensure that sigma values have a lower bound that is zero.

File: tests/testthat/test-piping-ini.R (hunk 2 above)
Comment (billdenney) [reply tests/testthat/test-piping-ini.R:435]: I added a simulation.

**Comment by billdenney** [comment]
The message already shows that the lower bound is changed with the normal `ini` code, so I didn't add anything specific there.  I think that I addressed all of the other review comments.

**Comment by billdenney** [comment]
@mattfidler, I think that this is ready to merge.

**Comment by mattfidler** [comment]
Thanks  @billdenney
